.github/workflows/test.yml

```diff
@@ -109,14 +109,14 @@ jobs:
           [
             default,
             no-default-features,
-            dns-over-rustls-aws-lc-rs,
-            dns-over-https-rustls-aws-lc-rs,
-            dns-over-quic-aws-lc-rs,
-            dns-over-h3-aws-lc-rs,
-            dns-over-rustls-ring,
-            dns-over-https-rustls-ring,
-            dns-over-quic-ring,
-            dns-over-h3-ring,
+            tls-aws-lc-rs,
```

Comment:
You can use `the ? syntax` to enable a combination of features. Use ring/aws-lc-rs as a global switch directly, without using tls-ring/https-ring or the like.

https://doc.rust-lang.org/cargo/reference/features.html#dependency-features

Reply:
This is the PR I submitted yesterday, you can refer to it.

https://github.com/mokeyish/self_update/blob/7f6052dbd85820e928362172a6e2ca317f435b84/Cargo.toml#L53

Reply:
I don't think that works well for these crates? If we want to make sure features are additive and basically all feature combinations do the right thing, then a protocol feature like just `quic` must chose either `aws-lc-rs` or `ring` implicitly, and that implicit dependency is exactly what I want to avoid.